design/aws-ecs/aws-ecs-tagging-support.md

```diff
@@ -0,0 +1,183 @@
+
+Amazon ECS supports tagging for all the ECS resources (see [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-using-tags.html)). Note that ECS provides a property of ECS service named `PropagateTags` to support propagating tags from either task definitions or services to tasks (tasks cannot be tagged using CDK without setting this parameter). Currently this property is not supported in CDK so that the tags are not propagated by default.
+
+In addition, `EnableECSManagedTags` is provided as another property so that ECS managed tags can be supported and used to tag ECS tasks. However, currently users cannot set `EnableECSManagedTags` and the default value is `false`.
+
+## General approach
+
+The new `BaseService` class will include two more base properties:
+
+* propagateTags
+* enableECSManagedTags
+
+`propagateTags` specifies whether to propagate the tags from the task definition or the service to ECS tasks and `enableECSManagedTags` specifies whether to enable Amazon ECS managed tags for the tasks within the service. Also, for `Ec2Service` and `FargateService` we will have the corresponding two new properties exposed to users:
+
+* propagateTaskTagsFrom (`SERVICE` | `TASK_DEFINITION` | `NONE`)
+* enableECSManagedTags (`true` | `false`)
+
+\*`SERVICE` and `true` are default values.\
+\*`propagateTaskTagsFrom` takes enum type `PropagatedTagSource`.\
+\*Note that the reason why we define `propagateTaskTagsFrom` is because we want to have a different name for `propagateTags` to eliminate the naming confusion.
+
+In addition, for the `aws-ecs-patterns` module we want it to be convenient, but in this case, the `aws-ecs-patterns` module is opinionated. As a result, we want `SERVICE` and `true` to be set by default, not just for constructs in the `aws-ecs-patterns` module.
+
+## Code changes
+
+Given the above, we should make the following changes on ECS:
+
+  1. Add an enum type `PropagatedTagSource` to support `propagateTaskTagsFrom`.
+  2. Add `propagateTags` and `enableECSManagedTags` properties to `BaseServiceOptions`.
+  3. Add `propagateTaskTagsFrom` properties to `Ec2ServiceProps` and `FargateServiceProps`.
+
+### Part 1: Add enum type `PropagatedTagSource` to support `propagateTaskTagsFrom`
+
+``` ts
+export enum PropagatedTagSource {
+  /**
+   * Propagate tags from service
+   */
+  SERVICE = 'SERVICE',
+
+  /**
+   * Propagate tags from task definition
+   */
+  TASK_DEFINITION = 'TASK_DEFINITION',
+
+  /**
+   * Do not propagate
+   */
+  NONE = 'NONE'
+}
+```
+
+### Part 2: Add `propagateTags` and `enableECSManagedTags` properties to `BaseServiceOptions`
+
+```ts
+export interface BaseServiceOptions {
+
+  ...
+
+   /**
+   * Specifies whether to propagate the tags from the task definition or the service to the tasks in the service
+   *
+   * Valid values are: PropagateTagFromType.SERVICE or PropagateTagFromType.TASK_DEFINITION
+   *
+   * @default none
+   */
+  readonly propagateTags?: PropagatedTagSource;
```

Comment:
Since you've added NONE to the enum, you should use that here.

Reply:
### done!